internal/backend/gs/gs.go

```diff
@@ -301,16 +317,45 @@ func (be *Backend) openReader(ctx context.Context, h restic.Handle, length int,
 
 	objName := be.Filename(h)
 
+	if be.encryptKey != "" {
+		encryptKeyPresent = true
+	}
+	for _, key := range be.decryptKeys {
+		if key != "" {
+			decryptKeysPresent = true
+			break
+		}
+	}
+
 	be.sem.GetToken()
 
-	r, err := be.bucket.Object(objName).NewRangeReader(ctx, offset, int64(length))
+	//https://cloud.google.com/storage/docs/encryption/using-customer-supplied-keys
+	err = errors.New(("Google Storage customer-supplied encryption keys failed."))
+	if encryptKeyPresent {
+		cryptoKey, _ = base64.StdEncoding.DecodeString(be.encryptKey)
```

Comment:
Please decode and verify the key once. That is the check should happen when the backend is opened and please also verify that the key is indeed as AES-256 key.

The same also applies to the decryption keys.

Reply:
Moved this calculation to the config loading stage.